src/sync_layer.rs

```diff
@@ -24,10 +24,17 @@ impl<T: Clone> GameStateCell<T> {
         state.checksum = checksum;
     }
 
-    /// Loads a `T` that the user previously saved into.
-    pub fn load(&self) -> Option<T> {
+    /// Provides direct (mutable) access to the `T` that the user previously saved into the cell.
+    ///
+    /// You probably want to use [Self::load()] instead; this function is useful only in niche use
+    /// cases.
+    ///
+    /// **Important**: the returned `T` must _not_ be modified in any way that affects (or may ever
```

Comment:
Is there an option to prevent the misuse of this? Any way of giving a non-mutable reference?

Reply:
I have pushed [eb1bebd](https://github.com/gschup/ggrs/pull/80/commits/eb1bebd2ed60beb46bfb37e26c5d44b8e06b5029) which implements the newtype pattern I mentioned in the PR's description (see commit description).

With that in place, the default is a shared reference, rather than a mutable reference - so possible misuse is not possible by accident (it has to be very explicitly opted into). It's a little more code but I like the result better than the original PR I raised; feel free to take it or leave it.

If you meant "is there an alternate parking_lot API we could use instead": I don't think it's possible to do this restriction without a newtype in our code, as there isn't any way to "unimplement" `DerefMut` for `MappedMutexGuard`, and there's no way I could see to downgrade a `MappedMutexGuard` to a hypothetical `SharedMutexGuard` either.